mockito-core/build.gradle.kts

```diff
@@ -200,3 +201,53 @@ fun mockitoExtensionConfigFile(project: Project, mockitoExtension: String) =
     file(project.layout.buildDirectory.dir("generated/resources/ext-config/test/mockito-extensions/$mockitoExtension"))
 //</editor-fold>
 
+//<editor-fold defaultstate="collapsed" desc="Generate version class file">
+// Manifest cannot be used because Android strips the manifest from the jar.
+
+val generateVersionClass by tasks.registering(GenerateVersionClass::class)
+
+sourceSets.main {
+    java {
+        srcDir(generateVersionClass)
+    }
+}
+
+tasks.withType<Checkstyle> {
+    exclude("org/mockito/internal/util/MockitoVersion.java")
+}
+
+configure<SpotlessExtension> {
+    java {
+        targetExclude(relativePath(generateVersionClass.get().generatedVersionClassDir) + "/**")
```

Comment:
Nit: to avoid eager configuration, let's also define a constant for `project.layout.buildDirectory.dir("generated/sources/version/java")`. That way, we define it once and use it twice to configure:

```kotlin
val generatedVersionClassDirLocation = project.layout.buildDirectory.dir("generated/sources/version/java")

generateVersionClass.configure {
    generatedVersionClassDir(generatedVersionClassDirLocation)
}

configure<SpotlessExtension> {
    java {
        targetExclude(relativePath(generatedVersionClassDirLocation) + "/**")
    }
}
```

That also allows us to remove the `.convention` on the task property

Reply:
Good observation!